sdk/translation/ai-translation-text/test/public/utils/recordedClient.ts

```diff
@@ -0,0 +1,115 @@
+  const endpoint =
+    env.TEXT_TRANSLATION_ENDPOINT || "https://fakeEndpoint.cognitive.microsofttranslator.com";
+  const apikey = env.TEXT_TRANSLATION_API_KEY || "fakeApikey";
+  const region = env.TEXT_TRANSLATION_REGION || "fakeRegion";
+
+  const translatorCredential = new TranslatorCredential(apikey, region);
+  const client = TextTranslationFactory(endpoint, translatorCredential, updatedOptions);
+  return client;
+}
+
+export async function createCustomTranslationClient(options: {
+  recorder?: Recorder;
+  clientOptions?: ClientOptions;
+}): Promise<TextTranslationClient> {
+  const { recorder, clientOptions = {} } = options;
+  const updatedOptions = recorder ? recorder.configureClientOptions(clientOptions) : clientOptions;
+  const customEndpoint =
+    env.TEXT_TRANSLATION_CUSTOM_ENDPOINT ||
+    "https://fakeCustomEndpoint.cognitiveservices.azure.com";
+  const apikey = env.TEXT_TRANSLATION_API_KEY || "fakeApikey";
+  const region = env.TEXT_TRANSLATION_REGION || "fakeRegion";
+
+  const translatorCredential = new TranslatorCredential(apikey, region);
+  const client = TextTranslationFactory(customEndpoint, translatorCredential, updatedOptions);
+  return client;
+}
+
+export async function createLanguageClient(options: {
+  recorder?: Recorder;
+  clientOptions?: ClientOptions;
+}): Promise<TextTranslationClient> {
+  const { recorder, clientOptions = {} } = options;
+  const updatedOptions = recorder ? recorder.configureClientOptions(clientOptions) : clientOptions;
+  const endpoint =
+    env.TEXT_TRANSLATION_ENDPOINT || "https://fakeEndpoint.cognitive.microsofttranslator.com";
+  const client = TextTranslationFactory(endpoint, undefined, updatedOptions);
+  return client;
+}
+
+export async function createTokenTranslationClient(options: {  
+  recorder?: Recorder;
+  clientOptions?: ClientOptions;
+}): Promise<TextTranslationClient> {
+  const { recorder, clientOptions = {} } = options;
+  const updatedOptions = recorder ? recorder.configureClientOptions(clientOptions) : clientOptions;
+  const endpoint = env.TEXT_TRANSLATION_ENDPOINT || "https://fakeEndpoint.cognitive.microsofttranslator.com";
+  const apikey = env.TEXT_TRANSLATION_API_KEY || "fakeApikey";
+  const region = env.TEXT_TRANSLATION_REGION || "fakeRegion"; 
+  
+  const issueTokenURL: string = "https://" + region + ".api.cognitive.microsoft.com/sts/v1.0/issueToken?Subscription-Key=" + apikey;
+  let credential: TokenCredential;
+  if (isPlaybackMode()) {
+    credential = createMockToken();
+  }
+  else {  
+    const tokenClient = createDefaultHttpClient();
+    const request = createPipelineRequest({
+      url: issueTokenURL,
+      method: "POST"
+    });
+    request.allowInsecureConnection = true;
+    const response = await tokenClient.sendRequest(request);
+    const token:string = response.bodyAsText!; 
+    credential = new StaticAccessTokenCredential(token);
```

Comment:
Could you please explain why we are getting the token manually like this instead of using azure/identity?

Reply:
The Cognitive Services token is not azure/identity token. It is created by special endpoint. And I believe azure/identity is feature we are calling AAD auth which is currently not supported by SDKs - planned to be added later.

Reply:
How do you expect your customers to use it?

Reply:
For now (preview), we need to find out whether the CS Token service is going to implement the SDK. Then customers would be using that. If there will not be SDK for that, we need to find long term solution.

Reply:
The [service supports AAD](https://learn.microsoft.com/en-us/azure/cognitive-services/translator/reference/v3-0-reference#authentication-with-azure-active-directory-azure-ad) and it is trivial to add AAD support in the SDK. It is exposed by the same interface `TokenCredential`. See https://github.com/Azure/azure-sdk-for-js/blob/09988fef7c7e3d4d03a9b8f21bb8f91a04a1732c/sdk/agrifood/agrifood-farming-rest/src/farmBeats.ts#L16

Reply:
We have not validated AAD for now. We plan to add it in later commit - if it is no-op in the code, then we will just update samples and other documentation.

Reply:
(not validated AAD in SDK, our service has AAD in prod already)

Reply:
Ok, noted. I am still concerned about the customer experience for this custom access token. Such tokens need care to refresh them. Could we just not support them until we get better SDK support for them?

Reply:
For GA, we will have long term solution.